### PR TITLE
feat: FILES-33 - Implement max version logic

### PR DESCRIPTION
**Pull request by RakuJa**
Add max number of version limit on keep forever and clone and upload versions.

Keep forever and clone return an error (too many versions) when reaching the maximum number of versions allowed

Upload versions instead allows upload of new versions, deleting always the oldest not kept forever version.

The number of maximum kept forever versions is the maximum number of versions - 2, this makes it so upload versions will never overwrite the latest version (the one that purge versions always keep).

### CODE REVIEW

**Comment by RakuJa** [comment]
Add max version check on clone
Delete only after writing with success the new version

File: core/src/main/java/com/zextras/carbonio/files/graphql/errors/GraphQLResultErrors.java [hunk 1]
```diff
@@ -259,4 +259,33 @@ public static GraphQLError linkNotFound(
       .build();
   }
 
+  /**
+   * This method generates an error when a requested node has reached the maximum number of
+   * versions. Besides from standard data, it adds the following custom fields in the extensions
+   * attribute:
+   * * <ul>
+   * *   <li>the errorCode for easily discriminating</li>
+   * *   <li>the id of the link not found</li>
```
Comment (federicorispo) [inline core/src/main/java/com/zextras/carbonio/files/graphql/errors/GraphQLResultErrors.java:268]: `link`?

**Review by federicorispo — APPROVED** [review]
Never `for` loop again

**Comment by RakuJa** [comment]
Refactor version count to repository, change current foreach to functional programming

**Pull request merged**